pyproject.toml

```diff
@@ -68,11 +68,13 @@ dev = [
     "ruff==0.9.10",
     "black[jupyter]",
     "poethepoet>=0.33.0",
-    "mypy>=1.0.0",
     { include-group = "build" },
     { include-group = "tests" },
     { include-group = "docs" },
 ]
+type-check = [
```

Comment:
could you move this group before the `dev` one, and add `{ include-group = "type-check" },` in `dev`? We need mypy to be installed as part of the dev dependencies